### PR TITLE
Use unix-compat instead of unix for windows compat

### PR DESCRIPTION
**Pull request by mightybyte**
Beginning of a better Windows compatibility story for Obelisk.

I have:

  - [X] Based work on latest `develop` branch
  - [X] Looked for lint in my changes with `hlint .` (lint found code you did not write can be left alone)
  - [X] Run the test suite: `$(nix-build -A selftest --no-out-link)`
  - [ ] (Optional) Run CI tests locally: `nix-build release.nix -A build.x86_64-linux --no-out-link` (or `x86_64-darwin` on macOS)

### CODE REVIEW

**Comment by 3noch** [comment]
@mightybyte have you tested this *on* windows?

**Comment by mightybyte** [comment]
Not yet.  (It's building now.)  I anticipate more obstacles before I can actually use an Obelisk app on Windows.  Things like jsaddle-wkwebview, etc.  My goal here is pretty narrow--to build an Obelisk app as a Windows desktop app with GHC.  Note that this doesn't necessarily include making the `ob` command line app work on Windows.  All I need is for the obelisk-provided libraries to build.  I also don't intend to tackle the problem of getting ghcjs to build on Windows at all.  So while this in no way gets to what's needed for full-blown Obelisk Windows support, it is a necessary precondition.

**Comment by 3noch** [comment]
Makes sense. As long as there's no expectation that I will start actively testing obelisk on Windows I don't mind accepting this provided nothing else suffers as a result.

**Comment by mightybyte** [comment]
Oh yes, I definitely don't expect any Windows testing.

File: lib/command/src/Obelisk/Command.hs [hunk 1]
```diff
@@ -363,7 +363,9 @@ main' argsCfg = do
         Just impl -> do
           -- Invoke the real implementation, using --no-handoff to prevent infinite recursion
           putLog Debug $ "Handing off to " <> T.pack impl
-          liftIO $ executeFile impl False ("--no-handoff" : myArgs) Nothing
+          --liftIO $ executeFile impl False ("--no-handoff" : myArgs) Nothing
```
Comment (3noch) [inline lib/command/src/Obelisk/Command.hs:366]:
```suggestion
```

**Pull request merged**